components/server/src/workspace/workspace-starter.ts

```diff
@@ -259,6 +260,21 @@ export class WorkspaceStarter {
         }
     }
 
+    protected getAdditionalImageAuth(projectEnvVars: ProjectEnvVarWithValue[]): Map<string, string> {
```

Comment:
I think you can "hydrate" these with values here (doing it even lower-level would be even better, but a bit more complicated, so here is fine).

```suggestion
    protected getAdditionalImageAuth(projectEnvVars: ProjectEnvVar[]): Map<string, string> {
        const projectEnvVarsWithValues = await this.projectDB.getProjectEnvironmentVariableValues(projectEnvVars);
```